packages/client/hmi-client/src/views/ModelRenderer.vue

```diff
@@ -0,0 +1,143 @@
+import { parsePetriNet2IGraph, NodeData, EdgeData, NodeType } from '@/services/modelRenderer';
+import * as d3 from 'd3';
+
+const MARKER_VIEWBOX = '-5 -5 10 10';
+const ARROW = 'M 0,-3.25 L 5 ,0 L 0,3.25';
+const pathFn = d3
+	.line<{ x: number; y: number }>()
+	.x((d) => d.x)
+	.y((d) => d.y);
+
+// This model can be deleted in future. Just used for init graph
+const model: PetriNet = {
+	T: [{ tname: 'inf' }, { tname: 'recover' }, { tname: 'death' }],
+	S: [{ sname: 'S' }, { sname: 'I' }, { sname: 'R' }, { sname: 'D' }],
+	I: [
+		{ it: 1, is: 1 },
+		{ it: 1, is: 2 },
+		{ it: 2, is: 2 },
+		{ it: 3, is: 2 }
+	],
+	O: [
+		{ ot: 1, os: 2 },
+		{ ot: 1, os: 2 },
+		{ ot: 2, os: 3 },
+		{ ot: 3, os: 4 }
+	]
+};
+class ModelPlanRenderer extends graphScaffolder.BasicRenderer<NodeData, EdgeData> {
+	setupDefs() {
+		// @ts-ignore
+		const svg = d3.select(this.svgEl);
+
+		// Clean up
+		svg.select('defs').selectAll('.edge-marker-end').remove();
+
+		// Arrow defs
+		svg
+			.select('defs')
+			.append('marker')
+			.classed('edge-marker-end', true)
+			.attr('id', 'arrowhead')
+			.attr('viewBox', MARKER_VIEWBOX)
+			.attr('refX', 2)
+			.attr('refY', 0)
+			.attr('orient', 'auto')
+			.attr('markerWidth', 15)
+			.attr('markerHeight', 15)
+			.attr('markerUnits', 'userSpaceOnUse')
+			.attr('xoverflow', 'visible')
+			.append('svg:path')
+			.attr('d', ARROW)
+			.style('fill', '#000')
+			.style('stroke', 'none');
+	}
+
+	renderNodes(selection: D3SelectionINode<NodeData>) {
+		const state = selection.filter((d) => d.data.type === NodeType.State);
+		const transitions = selection.filter((d) => d.data.type === NodeType.Transition);
+
+		transitions
+			.append('rect')
+			.classed('shape', true)
+			.attr('width', (d) => d.width)
+			.attr('height', (d) => d.height)
+			.style('fill', '#88C')
+			.style('stroke', '#888');
+
+		state
+			.append('circle')
+			.classed('shape', true)
+			.attr('cx', (d) => d.width * 0.5)
+			.attr('cy', (d) => d.height * 0.5)
+			.attr('r', (d) => d.width * 0.5)
+			.attr('fill', '#f80');
+
+		selection
+			.append('text')
+			.attr('y', -5)
+			.text((d) => d.label);
+	}
+
+	renderEdges(selection: D3SelectionIEdge<EdgeData>) {
+		selection
+			.append('path')
+			.attr('d', (d) => pathFn(d.points))
+			.style('fill', 'none')
+			.style('stroke', '#000')
+			.style('stroke-width', 2)
+			.attr('marker-end', 'url(#arrowhead)');
+	}
+}
+
+onMounted(async () => {
+	let renderer: ModelPlanRenderer | null = null;
+	const modelDrawnElement = document.getElementById('model-panel') as HTMLDivElement;
+	const g: IGraph<NodeData, EdgeData> = parsePetriNet2IGraph(model); // get graph from petri net representation
+
+	// @ts-ignore
+	renderer = new ModelPlanRenderer({
+		el: modelDrawnElement,
+		useAStarRouting: true,
+		runLayout: runDagreLayout
+	});
+
+	// Test on click
+	// @ts-ignore
+	renderer.on('node-click', (_eventName: string, _event, selection: D3SelectionINode<NodeData>) => {
```

Comment:
I think you can more these two `@ts-ignores`

Reply:
I removed them for cleanup